lib/stages/frbNetworkProcess.cpp

```diff
@@ -298,5 +228,335 @@ void frbNetworkProcess::main_thread() {
+            return -1;
+        }
+
+        // use a larger send buffer for the socket
+        const int n = 256 * 1024 * 1024;
+        if (setsockopt(sock_fd, SOL_SOCKET, SO_SNDBUF, (void*)&n, sizeof(n)) < 0) {
+            FATAL_ERROR("Network Thread: setsockopt() failed: %s ", strerror(errno));
+            return -1;
+        }
+
+        src_sockets.push_back({addr, sock_fd});
+    }
+
+    /* every node is introducing packets to the network. To achive load balancing a sequece_id is
+       computed for each node this will make sure none of the catalyst switches are overloaded with
+       traffic at a given point of time ofcourse this will be usefull when chrony is sufficuently
+       synchronized across all nodes..
+    */
+
+    int my_sequence_id =
+        (int)(my_node_id / 128) + 2 * ((my_node_id % 128) / 8) + 32 * (my_node_id % 8);
+    return my_sequence_id;
+}
+
+
+int frbNetworkProcess::initialize_destinations() {
+    // reading the L1 ip addresses from the config file
+    const std::vector<std::string> link_ip =
+        config.get<std::vector<std::string>>(unique_name, "L1_node_ips");
+    for (size_t i = 0; i < link_ip.size(); i++) {
+        sockaddr_in addr;
+        memset(&addr, 0, sizeof(addr));
+        addr.sin_family = AF_INET;
+        inet_pton(AF_INET, link_ip[i].c_str(), &addr.sin_addr);
+        if (!dest_sockets.count(addr.sin_addr.s_addr)) {
+            // new destination, initialize the entry in `dest_sockets`
+            addr.sin_port = htons(udp_frb_port_number);
+            int sending_socket = get_vlan_from_ip(link_ip[i].c_str()) - 6;
+            dest_sockets.insert(
+                {addr.sin_addr.s_addr, DestIpSocket{link_ip[i], addr, sending_socket}});
+        }
+        stream_dest.push_back(std::ref(dest_sockets.at(addr.sin_addr.s_addr)));
+    }
+    return link_ip.size();
+}
+
+void frbNetworkProcess::set_destination_active_callback(connectionInstance& conn,
+                                                        json& json_request) {
+    string host;
+    bool active;
+    try {
+        host = json_request["host"];
+        active = json_request["active"];
+    } catch (...) {
+        conn.send_error("Couldn't parse the request.", HTTP_RESPONSE::BAD_REQUEST);
+        return;
+    }
+    bool found = false;
+    for (auto& ip_dst : dest_sockets) {
+        auto& dst = std::get<1>(ip_dst);
+        if (dst.host == host) {
+            dst.active = active;
+            found = true;
+        }
+    }
+    if (found) {
+        INFO("Set destination {} active flag to {}", host, active);
+        conn.send_empty_reply(HTTP_RESPONSE::OK);
+    } else {
+        WARN("No such destination: {}", host);
+        conn.send_error("Couldn't parse the request.", HTTP_RESPONSE::BAD_REQUEST);
+    }
+}
+
+// internal utility functions for dealing with low-level sockets APIs
+/// Send a ping packet to the destination using source socket `s`
+static bool send_ping(int s, const DestIpSocket& dst);
```

Comment:
These comments aren't doxygen standard.   See: http://lwlab.dunlap.utoronto.ca/kotekan/dev/dev_style_guide_doc.html   I'm less fussy about internal helper functions, but this feels out of place style wise.